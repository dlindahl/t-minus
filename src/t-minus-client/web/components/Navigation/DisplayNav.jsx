import AppNavBar from './AppNavBar';
import SpawnPresenterWindowButton from '../Buttons/SpawnPresenterWindowButton';
import TeleprompterInput from '../Teleprompter/TeleprompterInput';

const DisplayNav = (props) => (
  <AppNavBar direction="row">
    <TeleprompterInput/>
    &nbsp;
    <SpawnPresenterWindowButton/>
  </AppNavBar>
);

export default DisplayNav;
