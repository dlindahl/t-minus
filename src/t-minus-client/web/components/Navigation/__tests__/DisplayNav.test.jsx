import configureStore from 'redux-mock-store'
import DisplayNav from '../DisplayNav'
import emptyObj from 'empty/object'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

const STATES = {
  default: {
    meta: emptyObj
  }
}
const store = configureStore()

describe('DisplayNav', () => {
  it('renders the component', () => {
    const component = renderer.create(
      <Provider store={store(STATES.default)}>
        <DisplayNav/>
      </Provider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
