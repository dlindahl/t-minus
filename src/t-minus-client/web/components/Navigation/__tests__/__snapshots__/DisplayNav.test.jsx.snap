exports[`DisplayNav renders the component 1`] = `
<nav
  style={
    Object {
      "background": "#2c3e50",
      "height": 50,
    }
  }>
  <ul
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "100%",
        "listStyleType": "none",
        "margin": 0,
        "padding": 0,
      }
    }>
    <li
      style={
        Object {
          "textAlign": "center",
        }
      }>
      <form
        onSubmit={[Function]}
        style={
          Object {
            "height": "100%",
            "textAlign": "left",
          }
        }>
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter text for teleprompter..."
          style={
            Object {
              "background": "transparent",
              "border": "none",
              "color": "#ecf0f1",
              "display": "inline-block",
              "fontSize": 18,
              "fontWeight": 200,
              "height": "100%",
              "maxWidth": "75vw",
              "minWidth": "25ch",
              "overflow": "hidden",
              "padding": 5,
              "width": "3ch",
            }
          } />
      </form>
    </li>
    <li
      style={
        Object {
          "flex": 1,
        }
      }>
      Â 
    </li>
    <li
      style={
        Object {
          "textAlign": "center",
        }
      }>
      <button
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "background": "transparent",
            "border": 0,
            "color": "#bdc3c7",
            "cursor": "pointer",
            "fontSize": "2rem",
            "height": 50,
            "lineHeight": 1,
            "padding": 0,
            "width": 50,
          }
        }>
        <svg
          fill="currentColor"
          fit={true}
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "verticalAlign": "top",
            }
          }
          viewBox="0 0 40 40"
          width="1em">
          <g>
            <path
              d="m23.4 5h11.6v11.6h-3.4v-5.9l-16.3 16.3-2.3-2.3 16.3-16.3h-5.9v-3.4z m8.2 26.6v-11.6h3.4v11.6q0 1.4-1 2.4t-2.4 1h-23.2q-1.4 0-2.4-1t-1-2.4v-23.2q0-1.4 1-2.4t2.4-1h11.6v3.4h-11.6v23.2h23.2z" />
          </g>
        </svg>
      </button>
    </li>
  </ul>
</nav>
`;
